// @vonage/messages/dist/cjs/classes/RCS/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/RCS/index.ts
var RCS_exports = {};
__export(RCS_exports, {
  RCSAbstractSuggestion: () => RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion: () => RCSCreateCalendarEventSuggestion,
  RCSCustom: () => RCSCustom,
  RCSDialActionSuggestion: () => RCSDialActionSuggestion,
  RCSFile: () => RCSFile,
  RCSImage: () => RCSImage,
  RCSOpenURLSuggestion: () => RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion: () => RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion: () => RCSReplySuggestion,
  RCSShareLocationSuggestion: () => RCSShareLocationSuggestion,
  RCSText: () => RCSText,
  RCSVideo: () => RCSVideo,
  RCSViewLocationSuggestion: () => RCSViewLocationSuggestion
});
module.exports = __toCommonJS(RCS_exports);

// lib/classes/AbstractMessage.ts
var AbstractMessage = class {
  /**
   * The recipient of the message.
   */
  to;
  /**
   * The sender of the message.
   */
  from;
  /**
   * An optional client reference for the message.
   */
  clientRef;
  /**
   * Specifies the URL to which Status Webhook messages will be sent for this
   * particular message. Over-rides account-level and application-level Status
   * Webhook url settings on a per-message basis.
   */
  webhookUrl;
  /**
   * Specifies which version of the Messages API will be used to send Status
   * Webhook messages for this particular message. For example, if v0.1 is
   * set, then the JSON body of Status Webhook messages for this message will
   * be sent in Messages v0.1 format. Over-rides account-level and
   * application-level API version settings on a per-message basis.
   */
  webhookVersion;
  /**
   * Constructs a new `AbstractMessage` instance.
   *
   * @param {MessageParams} params - The parameters for creating a message.
   */
  constructor(params) {
    this.to = params.to;
    this.from = params.from;
    this.clientRef = params.clientRef;
    this.webhookUrl = params.webhookUrl;
    this.webhookVersion = params.webhookVersion;
  }
};

// lib/classes/RCS/RCSCustom.ts
var RCSCustom = class extends AbstractMessage {
  /**
   * The channel for this message (always 'rcs').
   */
  channel = "rcs" /* RCS */;
  /**
   * The type of message (always 'custom').
   */
  messageType = "custom" /* CUSTOM */;
  /**
   * A custom payload. The schema of a custom object can vary widely.
   */
  custom;
  /**
   * The duration in seconds the delivery of a message will be attempted. By
   * default Vonage attempts delivery for 72 hours, however the maximum effective
   * value depends on the operator and is typically 24 - 48 hours. We recommend
   * this value should be kept at its default or at least 30 minutes.
   */
  ttl;
  /**
   * An object of optional settings for the RCS message.
   */
  rcs;
  /**
     * Sends a custom message through RCS
     *
     * @param {RCSCustomParams} params - The parameters for creating a RCS custom message.
     * @example
     * ```ts
     * import { RCSCustom } from '@vonage/messages';
     *
     * const { messageUUID } = await messagesClient.send(new RCSCustom({
     *  to: TO_NUMBER,
     *  from: FROM_NUMBER,
     *  custom: {
     *    foo: 'bar',
     *  }
     *  clientRef: 'my-personal-reference',
     * }));
     *
     * console.log(`Message sent successfully with UUID ${messageUUID}`);
     * ```
     */
  constructor(params) {
    super(params);
    this.custom = params.custom;
    this.ttl = params.ttl;
    if (params.rcs) {
      this.rcs = params.rcs;
    }
  }
};

// lib/classes/AbstractFileMessage.ts
var AbstractFileMessage = class extends AbstractMessage {
  /**
   * The type of message (always 'file').
   */
  messageType = "file" /* FILE */;
  file;
  /**
   * Constructs a new `AbstractFileMessage` instance.
   *
   * @param {MessageParamsFile} params - The parameters for creating a file message.
   */
  constructor(params) {
    super(params);
    this.file = params.file;
  }
};

// lib/classes/RCS/RCSFile.ts
var RCSFile = class extends AbstractFileMessage {
  /**
   * The channel for this message (always 'rcs').
   */
  channel = "rcs" /* RCS */;
  /**
   * The duration in seconds the delivery of a message will be attempted. By
   * default Vonage attempts delivery for 72 hours, however the maximum effective
   * value depends on the operator and is typically 24 - 48 hours. We recommend
   * this value should be kept at its default or at least 30 minutes.
   */
  ttl;
  /**
   * An object of optional settings for the RCS message.
   */
  rcs;
  /**
   * Sends a file message to the RCS channel.
   *
   * @param {RCSFileParams} params - The parameters for creating a RCS file message.
   *
   * @example
   * ```ts
   * import { RCSFile } from '@vonage/messages';
   *
   * const { messageUUID } = await messagesClient.send(new RCSFile({
   *  to: TO_NUMBER,
   *  from: FROM_NUMBER,
   *  file: {
   *    url: 'https://example.com/image.pdf',
   *  },
   *  clientRef: 'my-personal-reference',
   * }));
   *
   * console.log(`Message sent successfully with UUID ${messageUUID}`);
   * ```
   */
  constructor(params) {
    super(params);
    this.ttl = params.ttl;
    if (params.rcs) {
      this.rcs = params.rcs;
    }
  }
};

// lib/classes/AbstractImageMessage.ts
var AbstractImageMessage = class extends AbstractMessage {
  /**
   * The type of message (always 'image').
   */
  messageType = "image" /* IMAGE */;
  image;
  /**
   * Constructs a new `AbstractImageMessage` instance.
   *
   * @param {MessageParamsImage} params - The parameters for creating an image message.
   */
  constructor(params) {
    super(params);
    this.image = params.image;
  }
};

// lib/classes/RCS/RCSImage.ts
var RCSImage = class extends AbstractImageMessage {
  /**
   * The channel for this message (always 'rcs').
   */
  channel = "rcs" /* RCS */;
  /**
   * The duration in seconds the delivery of a message will be attempted. By
   * default Vonage attempts delivery for 72 hours, however the maximum effective
   * value depends on the operator and is typically 24 - 48 hours. We recommend
   * this value should be kept at its default or at least 30 minutes.
   */
  ttl;
  /**
   * An object of optional settings for the RCS message.
   */
  rcs;
  /**
   * Send an RCS image message.
   *
   * @param {RCSImageParams} params - The parameters for creating the image message.
   * @example
   * ```ts
   * import { RCSImage } from '@vonage/messages';
   *
   * const { messageUUID } = await messagesClient.send(new RCSImage({
   *  to: TO_NUMBER,
   *  from: FROM_NUMBER,
   *  image: {
   *    url: 'https://example.com/image.jpg',
   *  },
   *  clientRef: 'my-personal-reference',
   * }));
   *
   * console.log(`Message sent successfully with UUID ${messageUUID}`);
   * ```
   */
  constructor(params) {
    super(params);
    this.ttl = params.ttl;
    if (params.rcs) {
      this.rcs = params.rcs;
    }
  }
};

// lib/classes/AbstractTextMessage.ts
var AbstractTextMessage = class extends AbstractMessage {
  /**
   * The type of message (always 'text').
   */
  messageType = "text" /* TEXT */;
  /**
   * The text content of the message.
   */
  text;
  /**
   * Constructs a new `AbstractTextMessage` instance for text messages.
   *
   * @param {MessageParamsText} params - The parameters for creating a text message.
   */
  constructor(params) {
    super(params);
    this.text = params.text;
  }
};

// lib/classes/RCS/RCSText.ts
var RCSText = class extends AbstractTextMessage {
  /**
   * The channel for this message (always 'rcs').
   */
  channel = "rcs" /* RCS */;
  /**
   * The duration in seconds the delivery of a message will be attempted. By
   * default Vonage attempts delivery for 72 hours, however the maximum effective
   * value depends on the operator and is typically 24 - 48 hours. We recommend
   * this value should be kept at its default or at least 30 minutes.
   */
  ttl;
  /**
   * An array of suggestion objects to include with the message. You can
   * include up to 11 suggestions per message.
   */
  suggestions;
  /**
   * An object of optional settings for the RCS message.
   */
  rcs;
  /**
   * Sends a text message through the RCS channel.
   *
   * @param {RCSTextParams} params - The parameters for creating a RCS text message.
   *
   * @example
   * ```ts
   * import { RCSText } from '@vonage/messages';
   *
   * const { messageUUID } = await messagesClient.send(new RCSText({
   *  to: TO_NUMBER,
   *  from: FROM_NUMBER,
   *  text: 'Hello world',
   *  clientRef: 'my-personal-reference',
   * }));
   *
   * console.log(`Message sent successfully with UUID ${messageUUID}`);
   * ```
   */
  constructor(params) {
    super(params);
    this.ttl = params.ttl;
    this.suggestions = params.suggestions;
    if (params.rcs) {
      this.rcs = params.rcs;
    }
  }
};

// lib/classes/AbstractVideoMessage.ts
var AbstractVideoMessage = class extends AbstractMessage {
  /**
   * The type of message, which is 'video' for video messages.
   */
  messageType = "video" /* VIDEO */;
  /**
   * The video content of the message.
   */
  video;
  /**
   * Constructs a new `AbstractVideoMessage` instance for video messages.
   *
   * @param {MessageParamsVideo} params - The parameters for creating a video message.
   */
  constructor(params) {
    super(params);
    this.video = params.video;
  }
};

// lib/classes/RCS/RCSVideo.ts
var RCSVideo = class extends AbstractVideoMessage {
  /**
   * The channel for this message (always 'rcs').
   */
  channel = "rcs" /* RCS */;
  /**
   * The duration in seconds the delivery of a message will be attempted. By
   * default Vonage attempts delivery for 72 hours, however the maximum effective
   * value depends on the operator and is typically 24 - 48 hours. We recommend
   * this value should be kept at its default or at least 30 minutes.
   */
  ttl;
  /**
   * An object of optional settings for the RCS message.
   */
  rcs;
  /**
   * Send an RCS video message.
   *
   * @param {RCSVideoParams} params - The parameters for creating the video message.
   *
   * @example
   * ```ts
   * import { RCSVideo } from '@vonage/messages';
   *
   * const { messageUUID } = await messagesClient.send(new RCSVideo({
   *  to: TO_NUMBER,
   *  from: FROM_NUMBER,
   *  audio: {
   *    url: 'https://example.com/video.mp4',
   *  },
   *  clientRef: 'my-personal-reference',
   * }));
   *
   * console.log(`Message sent successfully with UUID ${messageUUID}`);
   * ```
   */
  constructor(params) {
    super(params);
    this.ttl = params?.ttl;
    if (params.rcs) {
      this.rcs = params.rcs;
    }
  }
};

// lib/classes/RCS/Suggestions/RCSAbstractSuggestion.ts
var RCSAbstractSuggestion = class {
  /**
   * The text shown on the suggestion chip and sent back if selected.
   */
  text;
  /**
   * Developer-defined opaque metadata to return in the inbound webhook
   * when the user selects this suggestion.
   */
  postbackData;
  constructor(params) {
    this.text = params.text;
    this.postbackData = params.postbackData;
  }
};

// lib/classes/RCS/Suggestions/RCSCreateCalendarEventSuggestion.ts
var RCSCreateCalendarEventSuggestion = class extends RCSAbstractSuggestion {
  type = "create_calendar_event" /* CREATE_CALENDAR_EVENT */;
  /**
   * The start time of the calendar event in ISO 8601 format.
   */
  startTime;
  /**
   * The end time of the calendar event in ISO 8601 format.
   */
  endTime;
  /**
   * The title of the calendar event
   */
  title;
  /**
   * A description of the calendar event
   */
  description;
  /**
   * A URL to open if the device is unable to display create a calendar event
   */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.title = params.title;
    this.startTime = params.startTime;
    this.endTime = params.endTime;
    this.description = params.description;
    this.fallbackUrl = params.fallbackUrl;
  }
};

// lib/classes/RCS/Suggestions/RCSDialActionSuggestion.ts
var RCSDialActionSuggestion = class extends RCSAbstractSuggestion {
  type = "dial" /* DIAL */;
  /**
   * The phone number to dial in E.164 format.
   */
  phoneNumber;
  /**
   * A URL to open if the device is unable to place a call.
   */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.phoneNumber = params.phoneNumber;
    this.fallbackUrl = params.fallbackUrl;
  }
};

// lib/classes/RCS/Suggestions/RCSOpenUrlSuggestion.ts
var RCSOpenURLSuggestion = class extends RCSAbstractSuggestion {
  type = "open_url" /* OPEN_URL */;
  /**
   * The URL to open when the suggestion is tapped.
   */
  url;
  /**
   * A short description of the URL for accessibility purposes.
   */
  description;
  constructor(params) {
    super(params);
    this.url = params.url;
    this.description = params.description;
  }
};

// lib/classes/RCS/Suggestions/RCSOpenUrlWebviewSuggestion.ts
var RCSOpenURLWebviewSuggestion = class extends RCSAbstractSuggestion {
  type = "open_url_in_webview" /* OPEN_URL_IN_WEBVIEW */;
  /**
   * The URL to open when the suggestion is tapped.
   */
  url;
  /**
   * A short description of the URL for accessibility purposes.
   */
  description;
  /**
   * The mode for displaying the URL in the webview window
   */
  viewMode;
  constructor(params) {
    super(params);
    this.url = params.url;
    this.description = params.description;
    this.viewMode = params.viewMode;
  }
};

// lib/classes/RCS/Suggestions/RCSReplySuggestion.ts
var RCSReplySuggestion = class extends RCSAbstractSuggestion {
  type = "reply" /* REPLY */;
  constructor(params) {
    super(params);
  }
};

// lib/classes/RCS/Suggestions/RCSShareLocationSuggestion.ts
var RCSShareLocationSuggestion = class extends RCSAbstractSuggestion {
  type = "share_location" /* SHARE_LOCATION */;
  constructor(params) {
    super(params);
  }
};

// lib/classes/RCS/Suggestions/RCSViewLocationSuggestion.ts
var RCSViewLocationSuggestion = class extends RCSAbstractSuggestion {
  type = "view_location" /* VIEW_LOCATION */;
  /**
   * The latitude of the location to view.
   */
  latitude;
  /**
   * The longitude of the location to view.
   */
  longitude;
  /**
  * A label to display on the location pin.
  */
  pinLabel;
  /**
  * A URL to open if the device is unable to display a map.
  */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.latitude = params.latitude;
    this.longitude = params.longitude;
    this.pinLabel = params.pinLabel;
    this.fallbackUrl = params.fallbackUrl;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion,
  RCSCustom,
  RCSDialActionSuggestion,
  RCSFile,
  RCSImage,
  RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion,
  RCSShareLocationSuggestion,
  RCSText,
  RCSVideo,
  RCSViewLocationSuggestion
});
