// @vonage/messages/dist/esm/classes/RCS/Suggestions/index.mjs
import "../../../chunk-LZRILNTJ.mjs";
import {
  RCSViewLocationSuggestion
} from "../../../chunk-CIGFWZCC.mjs";
import {
  RCSCreateCalendarEventSuggestion
} from "../../../chunk-5JYZSB7M.mjs";
import {
  RCSDialActionSuggestion
} from "../../../chunk-QJ64LPA6.mjs";
import {
  RCSOpenURLSuggestion
} from "../../../chunk-SA5UKHBB.mjs";
import {
  RCSOpenURLWebviewSuggestion
} from "../../../chunk-3A54LFFX.mjs";
import {
  RCSReplySuggestion
} from "../../../chunk-HSPIRCIP.mjs";
import {
  RCSShareLocationSuggestion
} from "../../../chunk-H3CMAMEH.mjs";
import {
  RCSAbstractSuggestion
} from "../../../chunk-SS2S4DEN.mjs";
import "../../../chunk-V645VKMB.mjs";
import "../../../chunk-EUOSRMKA.mjs";
import "../../../chunk-N4UOTIEF.mjs";
import "../../../chunk-SNU746A5.mjs";
import "../../../chunk-CN7QYYRQ.mjs";
import "../../../chunk-GTFM4XVI.mjs";
import "../../../chunk-4RITRMGD.mjs";
import "../../../chunk-57C2DYNH.mjs";
import "../../../chunk-SUPGMVFQ.mjs";
import "../../../chunk-UIMLGEV7.mjs";
import "../../../chunk-JWTAC4ZK.mjs";
import "../../../chunk-PGNE2O3U.mjs";
import "../../../chunk-QWOWKIZ6.mjs";
import "../../../chunk-75B4MJLH.mjs";
export {
  RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion,
  RCSDialActionSuggestion,
  RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion,
  RCSShareLocationSuggestion,
  RCSViewLocationSuggestion
};
