// @vonage/messages/dist/cjs/classes/RCS/Suggestions/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/RCS/Suggestions/index.ts
var Suggestions_exports = {};
__export(Suggestions_exports, {
  RCSAbstractSuggestion: () => RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion: () => RCSCreateCalendarEventSuggestion,
  RCSDialActionSuggestion: () => RCSDialActionSuggestion,
  RCSOpenURLSuggestion: () => RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion: () => RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion: () => RCSReplySuggestion,
  RCSShareLocationSuggestion: () => RCSShareLocationSuggestion,
  RCSViewLocationSuggestion: () => RCSViewLocationSuggestion
});
module.exports = __toCommonJS(Suggestions_exports);

// lib/classes/RCS/Suggestions/RCSAbstractSuggestion.ts
var RCSAbstractSuggestion = class {
  /**
   * The text shown on the suggestion chip and sent back if selected.
   */
  text;
  /**
   * Developer-defined opaque metadata to return in the inbound webhook
   * when the user selects this suggestion.
   */
  postbackData;
  constructor(params) {
    this.text = params.text;
    this.postbackData = params.postbackData;
  }
};

// lib/classes/RCS/Suggestions/RCSCreateCalendarEventSuggestion.ts
var RCSCreateCalendarEventSuggestion = class extends RCSAbstractSuggestion {
  type = "create_calendar_event" /* CREATE_CALENDAR_EVENT */;
  /**
   * The start time of the calendar event in ISO 8601 format.
   */
  startTime;
  /**
   * The end time of the calendar event in ISO 8601 format.
   */
  endTime;
  /**
   * The title of the calendar event
   */
  title;
  /**
   * A description of the calendar event
   */
  description;
  /**
   * A URL to open if the device is unable to display create a calendar event
   */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.title = params.title;
    this.startTime = params.startTime;
    this.endTime = params.endTime;
    this.description = params.description;
    this.fallbackUrl = params.fallbackUrl;
  }
};

// lib/classes/RCS/Suggestions/RCSDialActionSuggestion.ts
var RCSDialActionSuggestion = class extends RCSAbstractSuggestion {
  type = "dial" /* DIAL */;
  /**
   * The phone number to dial in E.164 format.
   */
  phoneNumber;
  /**
   * A URL to open if the device is unable to place a call.
   */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.phoneNumber = params.phoneNumber;
    this.fallbackUrl = params.fallbackUrl;
  }
};

// lib/classes/RCS/Suggestions/RCSOpenUrlSuggestion.ts
var RCSOpenURLSuggestion = class extends RCSAbstractSuggestion {
  type = "open_url" /* OPEN_URL */;
  /**
   * The URL to open when the suggestion is tapped.
   */
  url;
  /**
   * A short description of the URL for accessibility purposes.
   */
  description;
  constructor(params) {
    super(params);
    this.url = params.url;
    this.description = params.description;
  }
};

// lib/classes/RCS/Suggestions/RCSOpenUrlWebviewSuggestion.ts
var RCSOpenURLWebviewSuggestion = class extends RCSAbstractSuggestion {
  type = "open_url_in_webview" /* OPEN_URL_IN_WEBVIEW */;
  /**
   * The URL to open when the suggestion is tapped.
   */
  url;
  /**
   * A short description of the URL for accessibility purposes.
   */
  description;
  /**
   * The mode for displaying the URL in the webview window
   */
  viewMode;
  constructor(params) {
    super(params);
    this.url = params.url;
    this.description = params.description;
    this.viewMode = params.viewMode;
  }
};

// lib/classes/RCS/Suggestions/RCSReplySuggestion.ts
var RCSReplySuggestion = class extends RCSAbstractSuggestion {
  type = "reply" /* REPLY */;
  constructor(params) {
    super(params);
  }
};

// lib/classes/RCS/Suggestions/RCSShareLocationSuggestion.ts
var RCSShareLocationSuggestion = class extends RCSAbstractSuggestion {
  type = "share_location" /* SHARE_LOCATION */;
  constructor(params) {
    super(params);
  }
};

// lib/classes/RCS/Suggestions/RCSViewLocationSuggestion.ts
var RCSViewLocationSuggestion = class extends RCSAbstractSuggestion {
  type = "view_location" /* VIEW_LOCATION */;
  /**
   * The latitude of the location to view.
   */
  latitude;
  /**
   * The longitude of the location to view.
   */
  longitude;
  /**
  * A label to display on the location pin.
  */
  pinLabel;
  /**
  * A URL to open if the device is unable to display a map.
  */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.latitude = params.latitude;
    this.longitude = params.longitude;
    this.pinLabel = params.pinLabel;
    this.fallbackUrl = params.fallbackUrl;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion,
  RCSDialActionSuggestion,
  RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion,
  RCSShareLocationSuggestion,
  RCSViewLocationSuggestion
});
