// @vonage/messages/dist/cjs/classes/RCS/Suggestions/RCSCreateCalendarEventSuggestion.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/RCS/Suggestions/RCSCreateCalendarEventSuggestion.ts
var RCSCreateCalendarEventSuggestion_exports = {};
__export(RCSCreateCalendarEventSuggestion_exports, {
  RCSCreateCalendarEventSuggestion: () => RCSCreateCalendarEventSuggestion
});
module.exports = __toCommonJS(RCSCreateCalendarEventSuggestion_exports);

// lib/classes/RCS/Suggestions/RCSAbstractSuggestion.ts
var RCSAbstractSuggestion = class {
  /**
   * The text shown on the suggestion chip and sent back if selected.
   */
  text;
  /**
   * Developer-defined opaque metadata to return in the inbound webhook
   * when the user selects this suggestion.
   */
  postbackData;
  constructor(params) {
    this.text = params.text;
    this.postbackData = params.postbackData;
  }
};

// lib/classes/RCS/Suggestions/RCSCreateCalendarEventSuggestion.ts
var RCSCreateCalendarEventSuggestion = class extends RCSAbstractSuggestion {
  type = "create_calendar_event" /* CREATE_CALENDAR_EVENT */;
  /**
   * The start time of the calendar event in ISO 8601 format.
   */
  startTime;
  /**
   * The end time of the calendar event in ISO 8601 format.
   */
  endTime;
  /**
   * The title of the calendar event
   */
  title;
  /**
   * A description of the calendar event
   */
  description;
  /**
   * A URL to open if the device is unable to display create a calendar event
   */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.title = params.title;
    this.startTime = params.startTime;
    this.endTime = params.endTime;
    this.description = params.description;
    this.fallbackUrl = params.fallbackUrl;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSCreateCalendarEventSuggestion
});
