// @vonage/messages/dist/cjs/classes/RCS/Suggestions/RCSOpenUrlWebviewSuggestion.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/RCS/Suggestions/RCSOpenUrlWebviewSuggestion.ts
var RCSOpenUrlWebviewSuggestion_exports = {};
__export(RCSOpenUrlWebviewSuggestion_exports, {
  RCSOpenURLWebviewSuggestion: () => RCSOpenURLWebviewSuggestion
});
module.exports = __toCommonJS(RCSOpenUrlWebviewSuggestion_exports);

// lib/classes/RCS/Suggestions/RCSAbstractSuggestion.ts
var RCSAbstractSuggestion = class {
  /**
   * The text shown on the suggestion chip and sent back if selected.
   */
  text;
  /**
   * Developer-defined opaque metadata to return in the inbound webhook
   * when the user selects this suggestion.
   */
  postbackData;
  constructor(params) {
    this.text = params.text;
    this.postbackData = params.postbackData;
  }
};

// lib/classes/RCS/Suggestions/RCSOpenUrlWebviewSuggestion.ts
var RCSOpenURLWebviewSuggestion = class extends RCSAbstractSuggestion {
  type = "open_url_in_webview" /* OPEN_URL_IN_WEBVIEW */;
  /**
   * The URL to open when the suggestion is tapped.
   */
  url;
  /**
   * A short description of the URL for accessibility purposes.
   */
  description;
  /**
   * The mode for displaying the URL in the webview window
   */
  viewMode;
  constructor(params) {
    super(params);
    this.url = params.url;
    this.description = params.description;
    this.viewMode = params.viewMode;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSOpenURLWebviewSuggestion
});
