// @vonage/messages/dist/cjs/classes/RCS/Suggestions/RCSViewLocationSuggestion.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/RCS/Suggestions/RCSViewLocationSuggestion.ts
var RCSViewLocationSuggestion_exports = {};
__export(RCSViewLocationSuggestion_exports, {
  RCSViewLocationSuggestion: () => RCSViewLocationSuggestion
});
module.exports = __toCommonJS(RCSViewLocationSuggestion_exports);

// lib/classes/RCS/Suggestions/RCSAbstractSuggestion.ts
var RCSAbstractSuggestion = class {
  /**
   * The text shown on the suggestion chip and sent back if selected.
   */
  text;
  /**
   * Developer-defined opaque metadata to return in the inbound webhook
   * when the user selects this suggestion.
   */
  postbackData;
  constructor(params) {
    this.text = params.text;
    this.postbackData = params.postbackData;
  }
};

// lib/classes/RCS/Suggestions/RCSViewLocationSuggestion.ts
var RCSViewLocationSuggestion = class extends RCSAbstractSuggestion {
  type = "view_location" /* VIEW_LOCATION */;
  /**
   * The latitude of the location to view.
   */
  latitude;
  /**
   * The longitude of the location to view.
   */
  longitude;
  /**
  * A label to display on the location pin.
  */
  pinLabel;
  /**
  * A URL to open if the device is unable to display a map.
  */
  fallbackUrl;
  constructor(params) {
    super(params);
    this.latitude = params.latitude;
    this.longitude = params.longitude;
    this.pinLabel = params.pinLabel;
    this.fallbackUrl = params.fallbackUrl;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSViewLocationSuggestion
});
