// @vonage/messages/dist/cjs/enums/RCS/RCSCategory.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/RCS/RCSCategory.ts
var RCSCategory_exports = {};
__export(RCSCategory_exports, {
  RCSCategory: () => RCSCategory
});
module.exports = __toCommonJS(RCSCategory_exports);
var RCSCategory = /* @__PURE__ */ ((RCSCategory2) => {
  RCSCategory2["AUTHENTICATION"] = "authentication";
  RCSCategory2["TRANSACTION"] = "transaction";
  RCSCategory2["PROMOTION"] = "promotion";
  RCSCategory2["SERVICE_REQUEST"] = "service-request";
  RCSCategory2["ACKNOLEDGEMENT"] = "acknoledgement";
  return RCSCategory2;
})(RCSCategory || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSCategory
});
