// @vonage/messages/dist/cjs/enums/RCS/RCSSuggestionType.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/RCS/RCSSuggestionType.ts
var RCSSuggestionType_exports = {};
__export(RCSSuggestionType_exports, {
  RCSSuggestionType: () => RCSSuggestionType
});
module.exports = __toCommonJS(RCSSuggestionType_exports);
var RCSSuggestionType = /* @__PURE__ */ ((RCSSuggestionType2) => {
  RCSSuggestionType2["REPLY"] = "reply";
  RCSSuggestionType2["DIAL"] = "dial";
  RCSSuggestionType2["VIEW_LOCATION"] = "view_location";
  RCSSuggestionType2["SHARE_LOCATION"] = "share_location";
  RCSSuggestionType2["OPEN_URL"] = "open_url";
  RCSSuggestionType2["OPEN_URL_IN_WEBVIEW"] = "open_url_in_webview";
  RCSSuggestionType2["CREATE_CALENDAR_EVENT"] = "create_calendar_event";
  return RCSSuggestionType2;
})(RCSSuggestionType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RCSSuggestionType
});
