// @vonage/messages/dist/esm/classes/RCS/index.mjs
import "../../chunk-TG56TTBE.mjs";
import "../../chunk-LZRILNTJ.mjs";
import {
  RCSViewLocationSuggestion
} from "../../chunk-CIGFWZCC.mjs";
import {
  RCSCreateCalendarEventSuggestion
} from "../../chunk-5JYZSB7M.mjs";
import {
  RCSDialActionSuggestion
} from "../../chunk-QJ64LPA6.mjs";
import {
  RCSOpenURLSuggestion
} from "../../chunk-SA5UKHBB.mjs";
import {
  RCSOpenURLWebviewSuggestion
} from "../../chunk-3A54LFFX.mjs";
import {
  RCSReplySuggestion
} from "../../chunk-HSPIRCIP.mjs";
import {
  RCSShareLocationSuggestion
} from "../../chunk-H3CMAMEH.mjs";
import {
  RCSAbstractSuggestion
} from "../../chunk-SS2S4DEN.mjs";
import {
  RCSCustom
} from "../../chunk-YNEHMMVY.mjs";
import {
  RCSFile
} from "../../chunk-PA2FLXXQ.mjs";
import {
  RCSImage
} from "../../chunk-O6XXBX72.mjs";
import {
  RCSText
} from "../../chunk-UPOMCKCX.mjs";
import {
  RCSVideo
} from "../../chunk-L5M6ZE5W.mjs";
import "../../chunk-T4H7LUZQ.mjs";
import "../../chunk-YMRSCKGN.mjs";
import "../../chunk-JXK2KOJV.mjs";
import "../../chunk-LJPOB7FU.mjs";
import "../../chunk-V645VKMB.mjs";
import "../../chunk-EUOSRMKA.mjs";
import "../../chunk-N4UOTIEF.mjs";
import "../../chunk-SNU746A5.mjs";
import "../../chunk-CN7QYYRQ.mjs";
import "../../chunk-GTFM4XVI.mjs";
import "../../chunk-4RITRMGD.mjs";
import "../../chunk-57C2DYNH.mjs";
import "../../chunk-SUPGMVFQ.mjs";
import "../../chunk-UIMLGEV7.mjs";
import "../../chunk-JWTAC4ZK.mjs";
import "../../chunk-PGNE2O3U.mjs";
import "../../chunk-QWOWKIZ6.mjs";
import "../../chunk-75B4MJLH.mjs";
import "../../chunk-3WQKOHCE.mjs";
export {
  RCSAbstractSuggestion,
  RCSCreateCalendarEventSuggestion,
  RCSCustom,
  RCSDialActionSuggestion,
  RCSFile,
  RCSImage,
  RCSOpenURLSuggestion,
  RCSOpenURLWebviewSuggestion,
  RCSReplySuggestion,
  RCSShareLocationSuggestion,
  RCSText,
  RCSVideo,
  RCSViewLocationSuggestion
};
