// @vonage/server-client/dist/cjs/transformers.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/transformers.ts
var transformers_exports = {};
__export(transformers_exports, {
  camelCase: () => camelCase,
  camelCaseObjectKeys: () => camelCaseObjectKeys,
  isObject: () => isObject,
  kebabCase: () => kebabCase,
  kebabCaseObjectKeys: () => kebabCaseObjectKeys,
  omit: () => omit,
  snakeCase: () => snakeCase,
  snakeCaseObjectKeys: () => snakeCaseObjectKeys,
  transformObjectKeys: () => transformObjectKeys,
  words: () => words
});
module.exports = __toCommonJS(transformers_exports);
var words = (str) => str.split(/[-_\s]+/).flatMap((word) => word.split(/(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])/)).filter(Boolean);
var camelCase = (str) => words(str).map(
  (word, index) => index === 0 ? word.toLowerCase() : word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
).join("");
var snakeCase = (str) => words(str).map((word) => word.toLowerCase()).join("_");
var kebabCase = (str) => words(str).map((word) => word.toLowerCase()).join("-");
var isObject = (value) => {
  const type = typeof value;
  return value != null && (type === "object" || type === "function");
};
var transformObjectKeys = (transformFn, objectToTransform, deep = false, preserve = false) => {
  const transformedObject = {
    ...preserve ? objectToTransform : {}
  };
  for (const prop in objectToTransform) {
    if (!Object.prototype.hasOwnProperty.call(objectToTransform, prop)) {
      continue;
    }
    const newKey = transformFn(prop);
    const value = objectToTransform[prop];
    if (!deep || !isObject(value)) {
      transformedObject[newKey] = value;
      continue;
    }
    if (Array.isArray(value)) {
      transformedObject[newKey] = value.map(
        (t) => isObject(t) ? transformObjectKeys(
          transformFn,
          t,
          deep,
          preserve
        ) : t
      );
      continue;
    }
    transformedObject[newKey] = transformObjectKeys(
      transformFn,
      value,
      deep,
      preserve
    );
  }
  return transformedObject;
};
var camelCaseObjectKeys = (...rest) => transformObjectKeys(
  camelCase,
  ...rest
);
var snakeCaseObjectKeys = (...rest) => transformObjectKeys(
  snakeCase,
  ...rest
);
var kebabCaseObjectKeys = (...rest) => transformObjectKeys(
  kebabCase,
  ...rest
);
var omit = (keys, obj) => Object.fromEntries(
  Object.entries(obj).filter(([key]) => !keys.includes(key))
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  camelCase,
  camelCaseObjectKeys,
  isObject,
  kebabCase,
  kebabCaseObjectKeys,
  omit,
  snakeCase,
  snakeCaseObjectKeys,
  transformObjectKeys,
  words
});
